Osiris/Hacks/Visuals.cpp

```diff
@@ -946,6 +946,59 @@ void Visuals::updateShots(UserCmd* cmd) noexcept
         shotRecord.pop_front();
 }
 
+void Visuals::FootstepESP(GameEvent* event) noexcept
+{
+    if (!config->visuals.footsteps.footstepBeams.enabled)
+        return;
+
+    const auto entity = interfaces->entityList->getEntity(interfaces->engine->getPlayerForUserID(event->getInt("userid")));
+
+    if (!entity || !localPlayer.get() || entity == localPlayer.get() || entity->isDormant() || !entity->isAlive() || !entity->isOtherEnemy(localPlayer.get()))
+        return;
+
+    if (entity->getAbsOrigin() == localPlayer.get()->getAbsOrigin()) // fix for weird bug
+        return;
+
+    /*
+    @note: gr1ndy - other sprites that you can use:
+    "sprites/physbeam",
+    "sprites/purplelaser1",
+    "sprites/white.vmt", <-- draws behind the wall
+    */
+    
+    const auto modelIndex = interfaces->modelInfo->getModelIndex("sprites/purplelaser1.vmt");
+
+    BeamInfo info;
+
+    info.type = TE_BEAMRINGPOINT;
+    info.modelName = "sprites/purplelaser1.vmt";
+    info.modelIndex = modelIndex;
+    info.haloIndex = -1;
+    info.haloScale = 0.0f;
+    info.life = 2.0f;
+    info.width = static_cast<float>(config->visuals.footsteps.footstepBeamThickness);
+    info.fadeLength = 0.0f;
+    info.amplitude = 0.0f;
+    info.red = config->visuals.footsteps.footstepBeams.color[0] * 255;
+    info.green = config->visuals.footsteps.footstepBeams.color[1] * 255;
+    info.blue = config->visuals.footsteps.footstepBeams.color[2] * 255;
+    info.brightness = 255;
+    info.speed = 0.0f;
+    info.startFrame = 0;
+    info.frameRate = 60.0f;
+    info.segments = 1;
+    info.flags = FBEAM_FADEOUT;
+    info.ringCenter = entity->getAbsOrigin() + Vector(0.0f, 0.0f, 5.0f);
+    info.ringStartRadius = 0.0f;
+    info.ringEndRadius = static_cast<float>(config->visuals.footsteps.footstepBeamRadius);
```

Comment:
set sprites/white.vmt as default beam.

Reply:
> What's the point of using an int variable, then converting this to float?

To avoid a warning on compile

Reply:
> > What's the point of using an int variable, then converting this to float?
> 
> To avoid a warning

literally could use float and would be the same effect.